src/react-apps/applications/dashboard/src/services/fetchDashboardReducer.ts

```diff
@@ -0,0 +1,47 @@
+import update from 'immutability-helper';
+import { Action, Reducer } from 'redux';
+import * as FetchDashboardActions from './fetchDashboardActions';
+import * as FetchDashboardActionTypes from './fetchDashboardActionTypes';
+
+export interface IDashboardStoreState {
+  services: any;
+  user: any;
+
+}
+
+const initialState: IDashboardStoreState = {
+  services: [],
+  user: {},
+};
+
+const dashboardReducer: Reducer<IDashboardStoreState> = (
+  state: IDashboardStoreState = initialState,
+  action?: Action,
+): IDashboardStoreState => {
+  if (!action) {
+    return state;
+  }
+  switch (action.type) {
+    case FetchDashboardActionTypes.FETCH_SERVICES_FULFILLED: {
+      const { services } = action as FetchDashboardActions.IFetchServicesFulfilled;
+      return update<IDashboardStoreState>(state, {
+        $apply: () => ({
```

Comment:
because the IDashboardStoreState (renamed IDashboardState) contains both services and user the $appy operation works and the $set breaks the entire application

Reply:
Then use
```
update(state, {
   services: {
      $set: services,
   },
});
```